concrete/src/Http/Service/Ajax.php

```diff
@@ -2,11 +2,35 @@
 namespace Concrete\Core\Http\Service;
 
 use Exception;
+use Concrete\Core\Http\Request;
 
 class Ajax
 {
-    /** Sends a result to the client and ends the execution.
+    /**
+     * Check if a request is an Ajax call.
+     *
+     * @param Request $request
+     *
+     * @return bool
+     */
+    public function isAjaxRequest(Request $request)
+    {
+        $result = false;
+        $requestedWith = $request->server->get('HTTP_X_REQUESTED_WITH');
+        if (is_string($requestedWith) && strcasecmp($requestedWith, 'XMLHttpRequest') === 0) {
+            $result = true;
+        }
+
+        return $result;
+    }
+
+    /**
+     * Sends a result to the client and ends the execution.
+     *
      * @param mixed $result
+     *
+     * @deprecated You should switch to something like:
+     * return \Core::make(\Concrete\Core\Http\ResponseFactoryInterface::class)->json(...)
```

Comment:
FWIW I think it's best to recommend `$app->make` or $this->app->make` over `\Core`.